controllers/machinedeletionremediation_controller.go

```diff
@@ -270,6 +345,103 @@ func (r *MachineDeletionRemediationReconciler) setDeletedMachineAnnotation(ctx c
+	if err := r.Client.Status().Update(ctx, mdr); err != nil {
+		if !apiErrors.IsConflict(err) {
+			r.Log.Error(err, "failed to update mdr status")
+		}
+		return err
+	}
+	return nil
+}
+
+func (r *MachineDeletionRemediationReconciler) updateConditions(reason processingChangeReason, mdr *v1alpha1.MachineDeletionRemediation) error {
+	var processingConditionStatus, succeededConditionStatus metav1.ConditionStatus
+
+	switch reason {
+	case remediationStarted:
+		processingConditionStatus = metav1.ConditionTrue
+		succeededConditionStatus = metav1.ConditionUnknown
+	case remediationFinished:
+		processingConditionStatus = metav1.ConditionFalse
+		succeededConditionStatus = metav1.ConditionTrue
+	case remediationFailed:
+		processingConditionStatus = metav1.ConditionFalse
+		succeededConditionStatus = metav1.ConditionFalse
+	case remediationTerminatedByNHC:
+		processingConditionStatus = metav1.ConditionFalse
+		succeededConditionStatus = metav1.ConditionFalse
+	default:
+		err := fmt.Errorf("unknown processingChangeReason:%s", reason)
+		r.Log.Error(err, "couldn't update MDR Status Conditions")
+		return err
+	}
+
+	if meta.IsStatusConditionPresentAndEqual(mdr.Status.Conditions, v1alpha1.ProcessingConditionType, processingConditionStatus) &&
+		meta.IsStatusConditionPresentAndEqual(mdr.Status.Conditions, v1alpha1.SucceededConditionType, succeededConditionStatus) {
+		return nil
+	}
+
+	r.Log.Info("updating Status Condition", "processingConditionStatus", processingConditionStatus, "succededConditionStatus", succeededConditionStatus, "reason", string(reason))
+	meta.SetStatusCondition(&mdr.Status.Conditions, metav1.Condition{
+		Type:   v1alpha1.ProcessingConditionType,
+		Status: processingConditionStatus,
+		Reason: string(reason),
+	})
+
+	meta.SetStatusCondition(&mdr.Status.Conditions, metav1.Condition{
+		Type:   v1alpha1.SucceededConditionType,
+		Status: succeededConditionStatus,
+		Reason: string(reason),
+	})
+
+	return nil
+}
+
+func (r *MachineDeletionRemediationReconciler) getCurrentState(mdr *v1alpha1.MachineDeletionRemediation) machineDeletionRemediationState {
+	if meta.IsStatusConditionPresentAndEqual(mdr.Status.Conditions, v1alpha1.ProcessingConditionType, metav1.ConditionTrue) &&
+		meta.IsStatusConditionPresentAndEqual(mdr.Status.Conditions, v1alpha1.SucceededConditionType, metav1.ConditionUnknown) {
+
+		if annotations := mdr.GetAnnotations(); annotations != nil {
+			_, exists := annotations[MachineNameNamespaceAnnotation]
+			if exists {
+				return mdrStateDeletionRequested
+			}
+		}
+
+		return mdrStateStarted
+	}
+
+	if meta.IsStatusConditionPresentAndEqual(mdr.Status.Conditions, v1alpha1.ProcessingConditionType, metav1.ConditionFalse) {
+		if meta.IsStatusConditionPresentAndEqual(mdr.Status.Conditions, v1alpha1.SucceededConditionType, metav1.ConditionTrue) {
+			return mdrStateCompleted
+		}
+		return mdrStateFailed
+	}
+
+	return mdrStateNotStarted
```

Comment:
If `ProcessingConditionType` and `SuccededConditionType` are both `False`, the status should be `mdrStateFailed`.
For each remediation, `ProcessingConditionType` move to False only when completed, either with Success or Failure.
Did I miss other situations where `ProcessingConditionType` can be `False` without having been `True` before?

Reply:
> Did I miss other situations where ProcessingConditionType can be False without having been True before?

IIUC the condition has three values (true, false, and unknown), then potentially three are 3^2 (=9) options. While some might be not feasible. Below is a matrix for all the options, without removing the not feasible ones, from what I understand your code:

| ProcessingConditionType/SuccededConditionType |True |False|Unknown|
|---|---|---|---|
|True|mdrStateNotStarted| mdrStateNotStarted  | mdrStateStarted/mdrStateDeletionRequested |
|False| mdrStateCompleted  | mdrStateFailed  |  mdrStateFailed |
|Unknown|mdrStateNotStarted   |mdrStateNotStarted   |mdrStateNotStarted   |

Below is how I imagine it should be (the changes are in bold):
| ProcessingConditionType/SuccededConditionType |True |False|Unknown|
|---|---|---|---|
|True|mdrStateNotStarted (not feasible)| **mdrStateStarted/mdrStateDeletionRequested**  | mdrStateStarted/mdrStateDeletionRequested |
|False| mdrStateCompleted  | mdrStateFailed  |  mdrStateFailed |
|Unknown|mdrStateNotStarted   |mdrStateNotStarted   |mdrStateNotStarted   |

Reply:
What is the default values of the conditions? I didn't understand with which values they begin with...

Reply:
IIUC, it has three values once set, while at the beginning the condition is unset/not-present

Reply:
Looking again on **[updateConditions](https://github.com/medik8s/machine-deletion-remediation/pull/74/files#diff-e43973c0bb7cadba06a4ea1f15420fa0e4f203c1c9bea8405b0d8d8dacb4f3f3R356)** and **[getCurrentState](https://github.com/medik8s/machine-deletion-remediation/pull/74/files#diff-e43973c0bb7cadba06a4ea1f15420fa0e4f203c1c9bea8405b0d8d8dacb4f3f3R396)** functions I better understand the feasible cases for MDR. 
According to  updateConditions function there are only three combinations of the processing and succeeded conditions (`true&unknown = remediationStarted`, `false&true = remediationFinished`, and `false&false= remediationTerminatedByNHC, remediationFailed`), thus the other six cases from the above matrix are impossible to reach. 

My concern came from the [`if case`](https://github.com/medik8s/machine-deletion-remediation/pull/74/files#diff-e43973c0bb7cadba06a4ea1f15420fa0e4f203c1c9bea8405b0d8d8dacb4f3f3R410-R415) which could let other combinations of the processing and succeeded conditions to reach `mdrStateFailed` and `mdrStateNotStarted`. 
For more accuracy I would suggest to strict the "questionable" if, but it is only my preference.

Updated table/matrix of all the processing and succeeded conditions according to the above two functions:
| ProcessingConditionType/SuccededConditionType |True |False|Unknown|
|---|---|---|---|
|True|mdrStateNotStarted (not feasible)| mdrStateNotStarted  (not feasible) | mdrStateStarted/mdrStateDeletionRequested |
|False| mdrStateCompleted  | mdrStateFailed  |  mdrStateFailed  (not feasible) |
|Unknown|mdrStateNotStarted  (not feasible)  |mdrStateNotStarted  (not feasible)   |mdrStateNotStarted  (not feasible)  |

Reply:
The combination of condition values should not matter at all. Just look at a single condition, an set its value accordingly. Otherwise it could have been `Phase` 🤷🏼‍♂️ Nice read: https://github.com/kubernetes/kubernetes/issues/7856

Reply:
I agree, in fact I just need to look at `ProcessingCondition` and, if False, `SucceededCondition`
(going through the reading now, thanks)

Reply:
>  Just look at a single condition, an set its value accordingly

I am not sure I read your comment correctly, however, because in the function `getCurrentState` is not involved in setting the value of a state condition.